source/text/implementation/vss-string_vectors.adb

```diff
@@ -160,6 +185,21 @@ package body VSS.String_Vectors is
       raise Program_Error;
    end Read;
 
+   -------------
+   -- Replace --
+   -------------
+
+   procedure Replace
+     (Self  : in out Virtual_String_Vector'Class;
+      Index : Positive;
+      Item  : VSS.Strings.Virtual_String'Class) is
+   begin
```

Comment:
Self.Data can be null, while handler's Replace is defined with not null exclusion.

Check for null and for too small vector should be done here, and can do nothing in case of violation of precondition, but do it in safe way.